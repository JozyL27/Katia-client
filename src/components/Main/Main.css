.main__Animate-div {
    position: relative;
    z-index: 1;
    width: 80%;
    margin: 0 auto;
}

.main__Swipe {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
}

.main__Nav {
    width: 80%;
    height: 7vmax;
    max-height: 7rem;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    padding: 1vmax 0;
    margin: 0 auto;
}

.main__profile-button {
    height: 5vmax;
    max-height: 5rem;
}

.main__contacts-button {
    height: 5vmax;
    max-height: 5rem;
}

.main__Image {
    height: 15vmax;
    width: auto;
    max-width: 80%;
    margin: 1.5vmax 0;
    border-radius: 2.5vmax;
    transition: height .6s;
    transition-timing-function: ease-in-out;
}

.main__hidden-img {
    height: 0;
    margin: 0;
}

.main__Swipe-User {
    position: relative;
    min-height: 75vmin;
    height: 83vmax;
    max-height: 78vmax;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0 auto;
    border-radius: 2.5vmax;
    background: white;
    box-shadow: .5rem 1rem .8rem #888888;
    border: .1rem solid lightgrey;
}

.main__Swipe-User:hover {
    cursor: pointer;
}

.main__display-name {
    font-size: 5vmin;
    margin: 1vmin 0;
    max-width: 100%;
    text-align: center;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.main__bottom-card {
    position: relative;
    z-index: 2;
    width: 100%;
    height: 100%;
    max-height: 100%;
    padding-bottom: 4vmin;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    overflow-y: auto;
}

.main__card-header {
    width: 70%;
    font-size: 4vmin;
    font-weight: bold;
    padding: .8vh 2vw;
    margin: .8vh 0;
    transition: font-size .5s, height .5s, background-color .5s;
    transition-timing-function: ease-in-out;
    transition-delay: .1s;

    border: none;
    border-bottom: .2rem solid lightgrey;
    text-align: center;
    color: black;
}

.main__bottom-card > span {
    font-size: 3.5vmin;
    padding: .2vmin 0;
}

.main__bottom-card > .lfm-in {
    font-size: 3.5vmin;
    color: black;
}

.main__platforms {
    width: 100%;
    display: flex;
    justify-content: space-around;
}

.main__xbox, .main__playstation, .main__nintendo, .main__PC {
    height: 6vmin;
}

.main__none {
    font-size: 4vmin;
}

.main__genres {
    text-align: center;
    color: black;
}

.main__bio {
    font-size: 3.5vmin;
    text-align: center;
    padding: 0 1vw;
    max-height: 17vmax;
    transition: font-size .5s, height .5s, background-color .5s;
    transition-timing-function: ease-in-out;
    transition-delay: .1s;
    color: black;
    overflow-y: auto;
}

.main__hidden-text {
    font-size: 0;
    padding: 0;
    margin: 0;
    height: 0;
    transition-delay: 0s;
    border: none;
}

.main__caret-container {
    width: 100%;
    display: flex;
    justify-content: center;
}

.main__down-caret {
    position: absolute;
    bottom: 0.75vmin;
    height: 2.5vmin;
    transition: transform .5s;
    transition-timing-function: ease-in-out;
}

.reverse {
    transform: rotateX(180deg);
}

.main__Second-Nav {
    width: 80%;
    height: 10vmax;
    max-height: 12rem;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    margin: 0 auto;
    padding: 1vmax;
}

.main__x {
    height: 8vmax;
    max-height: 10rem;
}

.main__x:hover {
    cursor: pointer;
}

.main__check {
    height: 8vmax;
    max-height: 10rem;
}

.main__check:hover {
    cursor: pointer;
}

.main__Swipe-User-null {
    position: relative;
    height: 79vmax;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: .1rem solid lightgray;
    box-shadow: .5rem 1rem .8rem #888888;
    max-width: 80%;
    margin: 0 auto;
    border-radius: 2.5vmax;
}

.main__card-null {
    max-width: 80%;
    font-size: 5vmax;
    text-align: center;
}


.minViewInfo {
    display: flex;
    flex-direction: column;
    align-items: center;
    border: none;
    margin-top: -.2rem;
    width: calc(100% + .4rem);
    border-radius: 2.5vmax;
    padding: .5rem;
    background-color: white;
    box-shadow: 0 .8rem .8rem #888888;
    background: #30CFD0;
}

@media screen and (min-width: 600px) {
    .main__Swipe-User {
        height: 70vmax;
    }
    .main__Animate-div, .main__Nav, .main__Second-Nav, .main__Swipe-User-null {
        max-width: 70%;
    }
}

@media screen and (min-width: 900px) {
    .main__Swipe-User {
        height: 55vmax;
    }
    .main__Animate-div, .main__Nav, .main__Second-Nav, .main__Swipe-User-null {
        max-width: 55%;
    }

    .main__display-name {
        font-size: 4rem;
    }

    .main__card-header {
        font-size: 2.8rem;
    }

    .main__bottom-card > span {
        font-size: 2.8rem;
    }

    .main__bottom-card > .lfm-in {
        font-size: 2.8rem;
    }

    .main__xbox, .main__playstation, .main__nintendo, .main__PC {
        height: 5.5rem;
    }
    
    .main__none {
        font-size: 2.8rem;
    }

    .main__bio {
        font-size: 2.8rem;
    }

    .main__hidden-text {
        font-size: 0;
    }
}

@media screen and (min-width: 1080px) {
    .main__Swipe-User {
        height: 52vmax;
    }
}

@media screen and (min-width: 1440px) {
    .main__Swipe-User {
        height: 52vmax;
    }
}