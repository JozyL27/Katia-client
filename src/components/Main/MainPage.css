.mainImage {
    height: 200px;
}

.mainSwipeH2 {
    text-align: center;
}

.mainSwipeUser {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.mainNav {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    margin: 70px auto 30px auto;
}

.mainSwipeUser {
    border: 1px solid black;
    padding: 10px;
    max-width: 90%;
    margin: 0 auto;
    border-radius: 8px;
}

.secondNav {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    margin: 20px auto 30px auto;
}

@media screen and (min-width: 768px) {
    .mainImage {
        height: 320px;
    }

    .mainSwipeUser {
        max-width: 50%;
    }
}