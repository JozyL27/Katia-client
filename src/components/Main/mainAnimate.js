import React from 'react'

import {
    motion,
    useMotionValue,
    useTransform,
    AnimatePresence
  } from "framer-motion"
  
  export const MyComponent = (props) => {
    const x = useMotionValue(0)
    const background = useTransform(
      x,
      [-100, 0, 100],
      ["#C00", "#fff", "#0A0"]
    )
  
    return (
      <motion.div style={{ maxWidth: '80%', margin: 'auto' }}>
        <AnimatePresence>
        <motion.div
          drag="x"
          dragConstraints={{ left: 0, right: 0 }}
          style={{ x, border: `.5vmax solid`, 
          borderColor: background, 
          borderRadius: '7.2vmax', }}
          transition={{ duration: 0.5 }}
        >
        
        {props.children}

        </motion.div>
        </AnimatePresence>
      </motion.div>
    )
  }