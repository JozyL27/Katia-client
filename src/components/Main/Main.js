import React from 'react'
import './Main.css'
import UserContext from '../../Contexts/UserContext'
import SwipeService from '../../services/swipe-service'
import { Link } from 'react-router-dom'
import userSVG from '../../images/user.svg'
import contactsSVG from '../../images/contacts.svg'
import nintendoLogo from '../../images/nintendo_logo.png'
import playstationLogo from '../../images/playstation_color_png.png'
import PC_Logo from '../../images/PC_Keyboard_Mouse_Icon.png'
import xboxLogo from '../../images/xbox_logo_png.png'
import checkmarkSVG from '../../images/checkmark-circle-2.svg'
import x_markSVG from '../../images/x-circle.svg'
import down_caretSVG from '../../images/solid_caret-down.svg'
import Queue from '../../Utils/Queue';
import { Swipeable } from 'react-swipeable'
import { MyComponent } from './mainAnimate'


export default class MainPage extends React.Component {
    state = {
        queue: null,
        expanded: false,
        error: null
    }

    static contextType = UserContext;

    componentDidMount() {
        SwipeService.getPotentialMatches(this.context.user_id)
            .then(potentialMatches => {
                const queue = new Queue();
                potentialMatches.queue.forEach(match => {
                    queue.enqueue(match);
                });
                this.setState({ queue });
            })
            .catch(error => this.setState({error: error.message}));
    }

    toggleExpanded = () => {
        this.setState({ expanded: this.state.expanded ? false : true })
    }

    swipeLeft = () => {
        const { queue } = this.state;
        
        const rejection = queue.dequeue()

        SwipeService.addRejection(this.context.user_id, rejection.id)
        .then(() => {
            this.setState({ queue, expanded: false })
        })
        .catch(error => this.setState({ error }))
    }

    swipeRight = () => {
        const { queue } = this.state;

        const match = queue.dequeue();

        SwipeService.addMatch(this.context.user_id, match.id)
            .then(() => {
                this.setState({ queue, expanded: false });
            })
            .catch(error => this.setState({error}));
    }

    generateUserCard = (user) => {
        const handlers = {
            onSwipedLeft: () => this.swipeLeft(),
            onSwipedRight: () => this.swipeRight(),
            preventDefaultTouchmoveEvent: true,
            trackMouse: true,
            delta: 160,
        }

        return (
            <>
            <Swipeable {...handlers}>
                <MyComponent>
                <li className='main__Swipe-User' onClick={this.toggleExpanded}>
                    <div className='minViewInfo'>
                    {this.state.expanded ? (<img src={user.avatar} alt='avatar' 
                    className='main__Image main__hidden-img' />) : 
                    (<img src={user.avatar} alt='avatar' className='main__Image' />)}
                    <h3 className='main__display-name'>{user.display_name}</h3>
                    </div>
                    <h4 className='main__card-header'>Platforms</h4>
                    <div className='main__platforms'>
                        {user.platforms.includes("Xbox") ? <img className='main__xbox' src={xboxLogo} alt='Xbox logo' /> : null}
                        {user.platforms.includes("PlayStation") ? <img className='main__playstation' src={playstationLogo} alt='Playstation logo' /> : null}
                        {user.platforms.includes("Nintendo") ? <img className='main__nintendo' src={nintendoLogo} alt='Nintendo logo' /> : null}
                        {user.platforms.includes("PC") ? <img className='main__PC' src={PC_Logo} alt='PC logo' /> : null}
                    </div>
                    <h4 className='main__card-header'>LFM</h4>

                    {this.context.generateLfmElements(user.lfm_in)}

                    {/* <h4 className='main__card-header'>Genres</h4>
                    <span className='main__genres'>{this.context.generateGenreString(user.genres)}</span> */}

                    {this.state.expanded ? (<>
                    <h4 className='main__card-header'>Genres</h4>
                    <span className='main__genres'>{this.context.generateGenreString(user.genres)}</span>
                    <h4 className='main__card-header'>Bio</h4>
                    <p className='main__bio'>{user.bio}</p></>)
                    :
                    (<><h4 className='main__card-header main__hidden-text'>Bio</h4>
                    <p className='main__bio main__hidden-text'>{user.bio}</p></>)}
                    <div className='main__caret-container'>
                        <input className={`main__down-caret${this.state.expanded ? ' reverse' : ''}`} type="image" src={down_caretSVG} alt='down-caret' />
                    </div>
                </li>
                </MyComponent>
            </Swipeable>
            <div className='main__Second-Nav'>
                <img className='main__x' src={x_markSVG} alt='x' onClick={this.swipeLeft} />
                <img className='main__check' src={checkmarkSVG} alt='checkmark' onClick={this.swipeRight} />
            </div>
            </>
        )
    }

    generateNullCard = () => {
        return (
            <li className='main__Swipe-User-null'>
                <h4 className='main__card-null'>You've reached the end of the user queue! While you're waiting for potential matches, you may find use in this <a href='https://store.steampowered.com/tags/en/Singleplayer/'>list of games.</a></h4>
            </li>
        )
    }

    render() {
        const { queue } = this.state;

        if(queue == null) {
            return (
                <div className='loading'>loading</div>
            )
        }

        const userOne = queue.isEmpty() ? null : queue.peek();

        return (
            <section className='main__Swipe'>
                <div className='main__Nav'>
                    <Link to='/profile'>
                        <img className='main__profile-button' src={userSVG} alt='profile' />
                    </Link>
                    <Link to='/matches'>
                        <img className='main__contacts-button' src={contactsSVG} alt='contacts' />
                    </Link>
                </div>
                {userOne ? this.generateUserCard(userOne) : this.generateNullCard()}
            </section>
        )
    }
}