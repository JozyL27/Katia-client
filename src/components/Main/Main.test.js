import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Main from './Main'

describe('<Main />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Main />
      </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Main />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})