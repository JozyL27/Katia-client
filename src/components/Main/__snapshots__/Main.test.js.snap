// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders the UI as expected 1`] = `
<section
  className="main__Swipe"
>
  <div
    className="main__Nav"
  >
    <a
      href="/profile"
      onClick={[Function]}
    >
      <img
        alt="profile"
        className="main__profile-button"
        src="user.svg"
      />
    </a>
    <a
      href="/matches"
      onClick={[Function]}
    >
      <img
        alt="contacts"
        className="main__contacts-button"
        src="contacts.svg"
      />
    </a>
  </div>
  <li
    className="main__Swipe-User"
  >
    <img
      alt="avatar"
      className="main__Image"
    />
    <h3 />
    <h4>
      Platforms
    </h4>
    <span />
    <h4>
      LFM In
    </h4>
    <span />
    <img
      alt="down-caret"
      className="main__down-caret"
      src="solid_caret-down.svg"
    />
  </li>
  <div
    className="main__Second-Nav"
  >
    <img
      alt="x"
      className="main__x"
      src="x-circle.svg"
    />
    <img
      alt="checkmark"
      className="main__check"
      src="checkmark-circle-2.svg"
    />
  </div>
</section>
`;
